src/IO/WriteBufferFromS3.cpp

```diff
@@ -113,6 +113,7 @@ void WriteBufferFromS3::nextImpl()
             else
             {
                 file_segments.erase(file_segment_it, file_segments.end());
+                file_segment->complete(FileSegment::State::EMPTY);
```

Comment:
May be use here `State::PARTIALLY_DOWNLOADED_NO_CONTINUATION` instead, as it is used for the same case in `CachedReadBufferFromRemoteFS` (case when space reservation failed). So as a result no other thread will try to download this file segment. This segment state includes the case when nothing downloaded and not going to be downloaded.